actors/miner/src/vesting_state.rs

```diff
@@ -81,56 +139,78 @@ impl VestingFunds {
             Some(VestingFund { epoch: vest_epoch, amount: vest_this_time })
         });
 
-        // Take the old funds array and replace it with a new one.
-        let funds_len = self.funds.len();
-        let old_funds = mem::replace(&mut self.funds, Vec::with_capacity(funds_len));
+        let old_funds = self.load(store)?;
 
         // Fill back in the funds array, merging existing and new schedule.
-        self.funds.extend(
-            old_funds.into_iter().merge_join_by(new_funds, |a, b| a.epoch.cmp(&b.epoch)).map(
-                |item| match item {
-                    EitherOrBoth::Left(a) => a,
-                    EitherOrBoth::Right(b) => b,
-                    EitherOrBoth::Both(a, b) => {
-                        VestingFund { epoch: a.epoch, amount: a.amount + b.amount }
-                    }
-                },
-            ),
-        );
+        let mut new_funds = old_funds
+            .into_iter()
+            .merge_join_by(new_funds, |a, b| a.epoch.cmp(&b.epoch))
+            .map(|item| match item {
+                EitherOrBoth::Left(a) => a,
+                EitherOrBoth::Right(b) => b,
+                EitherOrBoth::Both(a, b) => {
+                    VestingFund { epoch: a.epoch, amount: a.amount + b.amount }
+                }
+            })
+            .peekable();
+
+        // Take any unlocked funds.
+        let unlocked =
+            new_funds.peeking_take_while(|fund| fund.epoch < current_epoch).map(|f| f.amount).sum();
```

Comment:
a utility function for this might be nice, it's repeated here and in `unlock_vested_funds`